io.go

```diff
@@ -6,16 +6,38 @@ import (
 	"io"
 )
 
-func Copy(ch chan<- []byte, r io.Reader) error {
+func Copy(ch chan<- []byte, r io.Reader, w io.Writer) error {
 	scanner := bufio.NewScanner(r)
 
 	for scanner.Scan() {
-		ch <- []byte(scanner.Text() + "\n")
+		line := scanner.Bytes()
+		cpy := make([]byte, len(line)+1)
+		copy(cpy, line)
+		line = append(line, '\n')
+
+		if err := writeFully(w, line); err != nil {
+			return err
+		}
+
+		ch <- []byte(line)
```

Comment:
`scanner.Bytes()` already returns a `[]byte`. So it looks like you're initializing a byte array from a byte array.
Is that necessary?